src/overlays/actors/ovl_Sample/ac_sample.c

```diff
@@ -1,54 +1,234 @@
 #include "ac_sample.h"
 #include "m_actor_dlftbls.h"
 #include "m_object.h"
+#include "m_rcp.h"
+#include "69E2C0.h"
+#include "m_msg_main.h"
 #include "overlays/gamestates/ovl_play/m_play.h"
 
-void Ac_Sample_Actor_ct(Actor* thisx, Game_Play* game_play);
-void Ac_Sample_Actor_dt(Actor* thisx, Game_Play* game_play);
-void Ac_Sample_Actor_main(Actor* thisx, Game_Play* game_play);
-void Ac_Sample_Actor_draw(Actor* thisx, Game_Play* game_play);
+void Ac_Sample_Actor_ct(Actor* thisx, Game_Play* play);
+void Ac_Sample_Actor_dt(Actor* thisx, Game_Play* play);
+void Ac_Sample_Actor_main(Actor* thisx, Game_Play* play);
+void Ac_Sample_Actor_draw(Actor* thisx, Game_Play* play);
 
-#if 0
-ActorProfile Sample_Profile = {
-    /* */ ACTOR_SAMPLE,
-    /* */ ACTOR_PART_4,
-    /* */ 0,
-    /* */ 0x9000,
-    /* */ OBJECT_12,
-    /* */ sizeof(Sample),
-    /* */ Ac_Sample_Actor_ct,
-    /* */ Ac_Sample_Actor_dt,
-    /* */ Ac_Sample_Actor_main,
-    /* */ Ac_Sample_Actor_draw,
-    /* */ NULL,
-};
-#endif
+// TODO: extract assets
+
+extern BaseSkeletonR cKF_bs_r_hnw;
+extern BaseAnimationR cKF_ba_r_hnw_move;
+
+extern u8 hnw_tmem_txt[];
+extern u16 hnw_face[];
+
+void Ac_Sample_ct_forCorect(Actor* thisx, Game_Play* play) {
+    static ClObjPipe_Init Ac_Sample_OcInfoData_forStand = {
+        { OC1_1 | OC1_TYPE_8 | OC1_TYPE_10 | OC1_TYPE_20, OC2_TYPE_20, COLSHAPE_PIPE },
+        { ELEM_FLAG_1 },
+        { { 0x1E, 0x3C, 0, { 0, 0, 0 } } },
+    };
+    Sample* this = (Sample*)thisx;
+
+    ClObjPipe_ct(play, &this->stand);
+    ClObjPipe_set5(play, &this->stand, thisx, &Ac_Sample_OcInfoData_forStand);
+}
+
+void Ac_Sample_Excute_Corect(Actor* thisx, Game_Play* play) {
+    UNUSED s32 pad;
+    UNUSED s32 pad2;
+    UNUSED s32 pad3;
+    Sample* this = (Sample*)thisx;
+    ClObjPipe* stand = &this->stand;
+
+    CollisionCheck_Uty_ActorWorldPosSetPipeC(thisx, stand);
+    CollisionCheck_setOC(play, &play->unk_2138, &stand->base);
+}
+
+void Ac_Sample_Actor_dt(Actor* thisx, Game_Play* play) {
+    Sample* this = (Sample*)thisx;
+    SkeletonInfoR* info = &this->skeletonInfo;
+
+    if (thisx->child != NULL) {
+        Actor_delete(thisx->child);
+    }
+
+    cKF_SkeletonInfo_R_dt(info);
+    ClObjPipe_dt(play, &this->stand);
+}
+
+void Ac_Sample_Animation_Base(Sample* this) {
+    cKF_SkeletonInfo_R_play(&this->skeletonInfo);
+}
+
+char str_mail[64];
```

Comment:
This should be in function static

Reply:
I was being cautious with static BSS due to BSS ordering in the long run, so do I just put it in function?

Reply:
Oh right, I forgot about bss. I guess it should be prefixed with Ac_Sample_ and keep it outside the function